### PR TITLE
[VL] Add dense_rank/percent_rank/cume_dist window functions support

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

* support dense_rank/percent_rank/cume_dist window functions.
* move window functions mapping to ExpressionMappings.
* merge some duplicated code.

## How was this patch tested?

add runQueryAndCompare in velox backend TestOperators.

### CODE REVIEW

**Comment by liujiayi771** [comment]
@rui-mo We need to add doWindowFunctionValidate function, some window functions may not be implemented by ch backend.

**Comment by rui-mo** [comment]
> @rui-mo We need to add doWindowFunctionValidate function, some window functions may not be implemented by ch backend.

@JkSelf Does Velox backend need that?

**Comment by JkSelf** [comment]
@liujiayi771 @rui-mo
Ch backend may need this validation and we can add the doWindowFunctionValidate  method  in [BackendSettings ](https://github.com/oap-project/gluten/blob/328efaeed2576a7bc93a823f3b809832445efb92/gluten-core/src/main/scala/io/glutenproject/backendsapi/BackendSettings.scala#L25) file to validate.  @zzcclp Do you have any suggestion?

**Comment by FelixYBW** [comment]
> FunctionValidate function, some window functions may not be implemented by c

Is there any UT to cover these functions? If so let's enable the UTs as well and check if Velox has the same output as Vanilla Spark

**Comment by liujiayi771** [comment]
> > FunctionValidate function, some window functions may not be implemented by c
> 
> Is there any UT to cover these functions? If so let's enable the UTs as well and check if Velox has the same output as Vanilla Spark

Yes, GlutenSQLQuerySuite contains window.sql and udf-window.sql tests, we already enable these UTs.

**Comment by liujiayi771** [comment]
@JkSelf All checks have passed, I will revert velox repo modify.

**Comment by zzcclp** [comment]
@lgbo-ustc @taiyang-li can you help to review this pr .

**Review by JkSelf — APPROVED** [review]
LGTM

**Comment by lgbo-ustc** [comment]
https://github.com/oap-project/gluten/pull/906/files

File: gluten-core/src/main/scala/io/glutenproject/expression/WindowFunctionsBuilder.scala [hunk 1]
```diff
@@ -19,21 +19,26 @@ package io.glutenproject.expression
...
-          ConverterUtils.ROW_NUMBER, Seq(rowNumber.dataType), FunctionConfig.OPT)
-      case rank: Rank =>
-        ConverterUtils.makeFuncName(
-          ConverterUtils.RANK, Seq(rank.dataType), FunctionConfig.OPT)
-      case other =>
-        throw new UnsupportedOperationException(s"not currently supported: $other.")
+    val substraitFunc = ExpressionMappings.window_functions_map.get(windowFunc.getClass)
+    if (substraitFunc.isDefined) {
+      val functionName = windowFunc match {
+        case _: RowNumber | _: Rank | _: DenseRank | _: PercentRank | _: CumeDist =>
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/expression/WindowFunctionsBuilder.scala:30]: Why we need this match? `ExpressionMappings.window_functions_map.get(windowFunc.getClass)` should have checked this

Comment (liujiayi771) [reply gluten-core/src/main/scala/io/glutenproject/expression/WindowFunctionsBuilder.scala]: You are right, I deleted the match.

**Comment by liujiayi771** [comment]
> https://github.com/oap-project/gluten/pull/906/files This pr handles the unsupported funcitons

These two PRs may conflict, I moved some constants defined in ConverterUtils to ExpressionMappings. We need to work together to resolve the conflict. In the long run, it is better to add a validator, velox does not support some window functions of CH.

**Comment by lgbo-ustc** [comment]
> > https://github.com/oap-project/gluten/pull/906/files This pr handles the unsupported funcitons
> 
> These two PRs may conflict, I moved some constants defined in ConverterUtils to ExpressionMappings. We need to work together to resolve the conflict. In the long run, it is better to add a native validator in [BackendSettings](https://github.com/oap-project/gluten/blob/328efaeed2576a7bc93a823f3b809832445efb92/gluten-core/src/main/scala/io/glutenproject/backendsapi/BackendSettings.scala#L25), velox does not support some window functions of CH.

Not too many conflicts, #906  adds a check for unsupported window functions.
Others look OK

**Review by jinchengchenghh — APPROVED** [review]
Wait for CI

**Pull request merged**